lab_1_keywords_tfidf/main.py

```diff
@@ -4,106 +4,138 @@
 """
 from typing import Optional, Union
 
+import math
+import string
+
 
 def clean_and_tokenize(text: str) -> Optional[list[str]]:
     """
     Removes punctuation, casts to lowercase, splits into tokens
-
     Parameters:
     text (str): Original text
-
     Returns:
     list[str]: A sequence of lowercase tokens with no punctuation
-
     In case of corrupt input arguments, None is returned
     """
-    pass
+    if not isinstance(text, str):
+        return None
+    for punc in string.punctuation:
+        text = text.replace(punc, '')
+        clean_words = text.lower().strip().split()
+    return clean_words
 
 
 def remove_stop_words(tokens: list[str], stop_words: list[str]) -> Optional[list[str]]:
     """
     Excludes stop words from the token sequence
-
     Parameters:
     tokens (List[str]): Original token sequence
     stop_words (List[str]: Tokens to exclude
-
     Returns:
     List[str]: Token sequence that does not include stop words
-
     In case of corrupt input arguments, None is returned
     """
-    pass
+    if not(isinstance(tokens, list) and tokens != [] and all(isinstance(token, str) for token in tokens)
```

Comment:
you can replace `if tokens != []` with just `if tokens` as in python empty objects are equal to False